tests/integrations/gcp/test_gcp.py

```diff
@@ -0,0 +1,320 @@
+        name = "projects/{}/locations/{}/functions/{}".format(
+            project_id, location_id, function_name
+        )
+
+        # STEP : Create a zip of cloud function
+
+        subprocess_kwargs = dict(subprocess_kwargs)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            main_py = os.path.join(tmpdir, "main.py")
+            with open(main_py, "w") as f:
+                f.write(code)
+
+            setup_cfg = os.path.join(tmpdir, "setup.cfg")
+
+            with open(setup_cfg, "w") as f:
+                f.write("[install]\nprefix=")
+
+            subprocess.check_call(
+                [sys.executable, "setup.py", "sdist", "-d", os.path.join(tmpdir, "..")],
+                **subprocess_kwargs
+            )
+
+            subprocess.check_call(
+                "pip install ../*.tar.gz -t .",
+                cwd=tmpdir,
+                shell=True,
+                **subprocess_kwargs
+            )
+            shutil.make_archive(os.path.join(tmpdir, "ball"), "zip", tmpdir)
+
+            # STEP : Generate a signed url
+            parent = "projects/{}/locations/{}".format(project_id, location_id)
+
+            api_request = (
+                functions_service.projects()
+                .locations()
+                .functions()
+                .generateUploadUrl(parent=parent)
+            )
+            upload_url_response = api_request.execute()
+
+            upload_url = upload_url_response.get("uploadUrl")
+
+            # STEP : Upload zip file of cloud function to generated signed url
+            with open(os.path.join(tmpdir, "ball.zip"), "rb") as data:
+                requests.put(
+                    upload_url,
+                    data=data,
+                    headers={
+                        "x-goog-content-length-range": "0,104857600",
+                        "content-type": "application/zip",
+                    },
+                )
+
+        # STEP : Create a new cloud function
+        location = "projects/{}/locations/{}".format(project_id, location_id)
+
+        function_url = "https://{}-{}.cloudfunctions.net/{}".format(
+            location_id, project_id, function_name
+        )
+
+        body = {
+            "name": name,
+            "description": "Created as part of testsuite for getsentry/sentry-python",
+            "entryPoint": "cloud_handler",
+            "runtime": functions_runtime,
+            "timeout": timeout,
+            "availableMemoryMb": 128,
+            "sourceUploadUrl": upload_url,
+            "httpsTrigger": {"url": function_url},
+        }
+
+        api_request = (
+            functions_service.projects()
+            .locations()
+            .functions()
+            .create(location=location, body=body)
+        )
+        api_request.execute()
+
+        # STEP : Invoke the cloud function
+        # Adding delay of 60 seconds for new created function to get deployed.
+        time.sleep(60)
```

Comment:
That is an awfully slow test for CI. Is there a way around this, such as polling the deployment status?

Reply:
I need to check if polling deployment status is possible or not. Although, GCP deployments do take an awful lot of time. I've checked and figured that this is a stable minimum time which is needed for the deployment to finish, (the minimum time which I observed was 45 seconds nothing less than that).
So, I'll check if it is possible to poll and verify the status of deployment, but the test cases will still be slow.

Reply:
Is there a way we could reuse existing deployments then perhaps?

Reply:
Well, that is something I haven't check yet and I'm not completely sure about this. 
Although, even if a small change is there in a particular file in cloud function we need to redeploy the code. That's the reason building this integration took so much more time as compared to AWS. I'm not completely aware if there is a separate process than this.